bson/son.py

```diff
@@ -63,7 +63,7 @@ def __delitem__(self, key):
         dict.__delitem__(self, key)
 
     def keys(self):
-        return list(self.__keys)
+        return OrderedDict.fromkeys(self.__keys)
```

Comment:
We want `keys` to return something compatible with what dict().keys() and OrderedDict().keys() return. We don't want to create a new OrderedDict here.

Reply:
Fixed.